src/main/java/me/totalfreedom/totalfreedommod/httpd/module/Module_logfile.java

```diff
@@ -0,0 +1,227 @@
+package me.totalfreedom.totalfreedommod.httpd.module;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import me.totalfreedom.totalfreedommod.TotalFreedomMod;
+import me.totalfreedom.totalfreedommod.admin.Admin;
+import me.totalfreedom.totalfreedommod.httpd.HTMLGenerationTools;
+import me.totalfreedom.totalfreedommod.httpd.HTTPDPageBuilder;
+import me.totalfreedom.totalfreedommod.httpd.HTTPDaemon;
+import me.totalfreedom.totalfreedommod.httpd.NanoHTTPD;
+import me.totalfreedom.totalfreedommod.httpd.NanoHTTPD.Response;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringEscapeUtils;
+import org.apache.commons.lang3.StringUtils;
+
+public class Module_logfile extends HTTPDModule
+{
+
+    private static final File LOG_FOLDER = new File("./logs/");
+    private static final String[] LOG_FILTER = new String[]
+    {
+        "log",
+        "gz"
+    };
+
+    public Module_logfile(TotalFreedomMod plugin, NanoHTTPD.HTTPSession session)
+    {
+        super(plugin, session);
+    }
+
+    @Override
+    public Response getResponse()
+    {
+        try
+        {
+            return new HTTPDPageBuilder(body(), title(), null, null).getResponse();
+        }
+        catch (ResponseOverrideException ex)
+        {
+            return ex.getResponse();
+        }
+    }
+
+    public String title()
+    {
+        return "TotalFreedomMod :: Logfiles";
+    }
+
+    public String body() throws ResponseOverrideException
+    {
+        if (!LOG_FOLDER.exists())
+        {
+            return HTMLGenerationTools.paragraph("Can't find the logs folder.");
+        }
+
+        final StringBuilder out = new StringBuilder();
+        final String remoteAddress = socket.getInetAddress().getHostAddress();
+        final String[] args = StringUtils.split(uri, "/");
+        final ModuleMode mode = ModuleMode.getMode(getArg(args, 1));
+
+        switch (mode)
+        {
+            case LIST:
+            {
+                if (!isAuthorized(remoteAddress))
+                {
+                    out.append(HTMLGenerationTools.paragraph("Log files access denied: Your IP, " + remoteAddress + ", is not registered to a superadmin on this server."));

```

Comment:
This should also be logged server side as well to keep a log of who is attempting to access the logs.